backend/src/main/java/asegroup1/api/models/landregistry/LandRegistryQuery.java

```diff
@@ -0,0 +1,150 @@
+package asegroup1.api.models.landregistry;
+
+import java.util.List;
+
+public class LandRegistryQuery implements LandRegistryQueryBody {
+
+	private LandRegistryQueryBody body;
+
+	private LandRegistryQueryGroup groupConstraint;
+
+	private LandRegistryQuerySelect select;
+
+
+	public LandRegistryQuery(LandRegistryQueryBody body, LandRegistryQueryGroup groupConstraint, LandRegistryQuerySelect select) {
+		this.body = body;
+		this.groupConstraint = groupConstraint;
+		this.select = select;
+	}
+
+	public LandRegistryQuery() {
+		body = new LandRegistryQueryConstraint();
+		groupConstraint = new LandRegistryQueryGroup();
+		select = new LandRegistryQuerySelect();
+	}
+
+	public LandRegistryQueryBody getBody() {
+		return body;
+	}
+
+	public void setBody(LandRegistryQueryBody body) {
+		this.body = body;
+	}
+
+	public LandRegistryQueryGroup getGroupConstraint() {
+		return groupConstraint;
+	}
+
+	public void setGroupConstraint(LandRegistryQueryGroup groupConstraint) {
+		this.groupConstraint = groupConstraint;
+	}
+
+	public LandRegistryQuerySelect getSelect() {
+		return select;
+	}
+
+	public void setSelect(LandRegistryQuerySelect select) {
+		this.select = select;
+	}
+
+	public void addVarToGroup(String varname, boolean addToSelect) {
+		getGroupConstraint().select(varname);
+		if (addToSelect) {
+			getSelect().addSelectValue(varname);
+		}
+	}
+
+	public void addVarToGroup(String varname) {
+		addVarToGroup(varname, false);
+	}
+
+	public void addVarToSelect(String referenceName, Aggrigation aggrigation, String aggrigationResult) {
+		getSelect().addSelectValue(referenceName, aggrigation, aggrigationResult);
+	}
+
+	@Override
+	public String buildQueryContent() {
+		boolean grouping = !(groupConstraint == null || groupConstraint.getSelectables().isEmpty());
+		StringBuilder queryBuilder = new StringBuilder(select.buildQuerySelect(!grouping));
+		queryBuilder.append("\n");
+		queryBuilder.append("WHERE { \n\t" + body.buildQueryContent().replace("\n", "\n\t") + "\n}");
```

Comment:
This code can be cleaned up to the following

```StringBuilder queryBuilder =
                new StringBuilder(select.buildQuerySelect(!grouping))
                        .append("\n")
                        .append("WHERE { \n\t")
                        .append(body.buildQueryContent().replace("\n", "\n\t"))
                        .append("\n}");
        if (grouping) {
            queryBuilder
                    .append("\n")
                    .append(groupConstraint.buildGroup());```

Also, string concatenation should not take place inside the append argument. Instead, use an additional append.